### PR TITLE
Export powerful feature name and use in spec

### LINKED ISSUE #145: "compute pressure" powerful feature should probably be exported
See w3c/accelerometer#67 for example. Something similar should probably be done here.

### PR DESCRIPTION
**Pull request by kenchris**
Fixed #145

***
<a href="https://pr-preview.s3.amazonaws.com/kenchris/compute-pressure/pull/148.html" title="Last updated on Nov 14, 2022, 10:08 AM UTC (28bfc1f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/compute-pressure/148/3849d19...kenchris:28bfc1f.html" title="Last updated on Nov 14, 2022, 10:08 AM UTC (28bfc1f)">Diff</a>

### CODE REVIEW

**Comment by kenchris** [comment]
@rakuco please have a look

**Comment by rakuco** [comment]
@miketaylr could you double check this one? ReSpec doesn't seem to have anything similar to Bikeshed's `<dfn permission>` so I wonder if just exporting this is enough.

**Comment by kenchris** [comment]
Should we just land this for now as there are other good changes? We can reopen the issue when/if @miketaylr responds

**Comment by miketaylr** [comment]
> @miketaylr could you double check this one? ReSpec doesn't seem to have anything similar to Bikeshed's `<dfn permission>` so I wonder if just exporting this is enough.

Whoops, missed the ping due to some travel. Digging out of my inbox now... 🙈

File: index.html [hunk 1]
```diff
@@ -497,18 +498,18 @@ <h3>The <dfn>observe()</dfn> method</h3>
         </li>
         <li>
           If |document| is not null and is not [=allowed to use=] the [=policy-controlled
-          feature=] named "`compute-pressure`", throw {{NotAllowedError}}.
+          feature=] named "<a>compute-pressure</a>", throw {{NotAllowedError}}.
```
Comment (miketaylr) [inline index.html:501]: same nit about linking to the powerful feature for permissions-policy integration.

**Review by miketaylr — CHANGES_REQUESTED** [review]

File: index.html [hunk 2]
```diff
@@ -249,7 +249,8 @@ <h3>
     </h3>
     <p>
       The Compute Pressure API is a [=powerful feature=] which is identified
-      by the name "`compute-pressure`".
+      by the [=powerful feature/name=] "<dfn class="export">compute-pressure</dfn>",
```
[the remark this answers is not in the document]
Comment (rakuco) [reply index.html:252]: Ah, nice, that's what I was looking for. Do you know what needs to be done for that to be added to the main respec.org docs?

Comment (kenchris) [inline index.html:252]:
```suggestion
      by the [=powerful feature/name=] "<dfn class="permission export">compute-pressure</dfn>",
```

File: index.html [hunk 3]
```diff
@@ -260,8 +261,8 @@ <h3>
   </h3>
   <p>
     The Compute Pressure API defines a [=policy-controlled feature=]
-    identified by the string `"compute-pressure"`. Its [=default
-    allowlist=] is `["self"]`.
+    identified by [=powerful feature/name=] "<a>compute-pressure</a>".
```
[the remark this answers is not in the document]
Comment (kenchris) [reply index.html:264]: Here the powerful feature is also a permission policy controlled feature, so I thought it made sense to treat them as one and the same given the below:

image

File: index.html (hunk 1 above)
Comment (kenchris) [reply index.html:501]: See above

File: index.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (miketaylr) [reply index.html]: I don't, but I asked in https://github.com/w3c/respec/issues/4296#issuecomment-1307975697

File: index.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (miketaylr) [reply index.html:264]: Yeah, it's a bit confusing. Powerful features are identified by their name (https://w3c.github.io/permissions/#dfn-name), but permissions policies are identified by tokens. 

Screen Shot 2022-11-09 at 9 20 58 AM

Technically they're both just strings at the end of the day, so it doesn't really change anything implementation- or developer-wise. But I think it's unspecified what happens when you pass a powerful feature name to the permissions policy algorithms. 🤷

Comment (kenchris) [inline index.html:264]:
```suggestion
    identified by the token "<a>compute-pressure</a>".
```

File: index.html (hunk 1 above)
Comment (kenchris) [inline index.html:501]:
```suggestion
          feature=] token "compute-pressure", throw {{NotAllowedError}}.
```

**Comment by rakuco** [comment]
Note that ReSpec is complaining about the formatting here:

```
Invalid permission "compute-pressure".
How to fix: Check that the permission is quoted with double quotes.

Occurred 1 times at:

[<dfn>](https://wicg.github.io/compute-pressure/#respec-offender-invalid-permission-compute-pressure) element
```

**Comment by rakuco** [comment]
You could try following the format used in the ReSpec example: https://respec.org/docs/#permission

**Pull request merged**